pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -686,10 +700,11 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 		}
 		// For inbound, we only use EDS for the VIP cases. The VIP cluster will point to encap listener.
 		if supportsTunnel {
-			address := e.Address
+			addresses = e.Addresses
 			// We will connect to CONNECT origination internal listener, telling it to tunnel to ip:15008,
 			// and add some detunnel metadata that had the original port.
-			ep.Metadata.FilterMetadata[util.OriginalDstMetadataKey] = util.BuildTunnelMetadataStruct(address, int(e.EndpointPort), waypoint)
+			// TODO: support multiple addresses, currently only generate the Metadata once based on the first address.
```

Comment:
can address have different metadata?

If so, the design is fundamentally broken.

If not, no TODO needed.

Reply:
Okay, prefer to that all addresses should have the same metadata, remove the `TODO` description.